test/integration/resourcemanager/nodeagentauthorizer/nodeagentauthorizer_suite_test.go

```diff
@@ -0,0 +1,244 @@
+	testNamespaceName := testID + "-" + testRunID[:8]
+
+	By("Start source environment")
+	sourceEnv = &envtest.Environment{
+		CRDInstallOptions: envtest.CRDInstallOptions{
+			Paths: []string{
+				filepath.Join("..", "..", "..", "..", "example", "seed-crds", "10-crd-machine.sapcloud.io_machines.yaml"),
+			},
+		},
+		ErrorIfCRDPathMissing: true,
+	}
+
+	var err error
+	sourceRestConfig, err = sourceEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(sourceRestConfig).NotTo(BeNil())
+
+	DeferCleanup(func() {
+		By("Stop source environment")
+		Expect(sourceEnv.Stop()).To(Succeed())
+	})
+
+	By("Create source client")
+	sourceClient, err = client.New(sourceRestConfig, client.Options{Scheme: resourcemanagerclient.CombinedScheme})
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Create test Namespace")
+	testNamespace = &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			// create dedicated namespace for each test run, so that we can run multiple tests concurrently for stress tests
+			Name: testNamespaceName,
+		},
+	}
+	Expect(sourceClient.Create(ctx, testNamespace)).To(Succeed())
+	log.Info("Created test Namespace in source cluster", "namespaceName", testNamespace.Name)
+
+	DeferCleanup(func() {
+		By("Delete test Namespace from source cluster")
+		Expect(sourceClient.Delete(ctx, testNamespace)).To(Or(Succeed(), BeNotFoundError()))
+	})
+
+	By("Create kubeconfig file for source cluster")
+	kubeconfigFileName, err := createKubeconfigFileForAuthorizationWebhook(sourceEnv.WebhookInstallOptions)
+	Expect(err).ToNot(HaveOccurred())
+	DeferCleanup(func() {
+		By("Delete kubeconfig file for source cluster")
+		Expect(os.Remove(kubeconfigFileName)).To(Succeed())
+	})
+
+	By("Start target environment")
+	Expect(framework.FileExists(kubeconfigFileName)).To(BeTrue())
+	testAPIServer := &envtest.APIServer{}
+	testAPIServer.Configure().
+		Set("authorization-mode", "RBAC", "Webhook").
+		Set("authorization-webhook-config-file", kubeconfigFileName).
+		Set("authorization-webhook-cache-authorized-ttl", "0").
+		Set("authorization-webhook-cache-unauthorized-ttl", "0")
+
+	targetEnv = &envtest.Environment{
+		ControlPlane: envtest.ControlPlane{
+			APIServer: testAPIServer,
+		},
+	}
+
+	targetRestConfig, err = targetEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(targetRestConfig).NotTo(BeNil())
+
+	DeferCleanup(func() {
+		By("Stop target environment")
+		Expect(targetEnv.Stop()).To(Succeed())
+	})
+
+	By("Create target clients")
+	targetClient, err = client.New(targetRestConfig, client.Options{Scheme: resourcemanagerclient.CombinedScheme})
+	Expect(err).NotTo(HaveOccurred())
+
+	machineName = "machine-" + testRunID
+	userNameNodeAgent = "gardener.cloud:node-agent:machine:" + machineName
+
+	user, err := targetEnv.AddUser(
+		envtest.User{Name: userNameNodeAgent, Groups: []string{v1beta1constants.NodeAgentsGroup}},
+		&rest.Config{QPS: 1000.0, Burst: 2000.0},
+	)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(user).NotTo(BeNil())
```

Comment:
I don't think we need this line? If `err` has not occurred, `user` is never `nil`

Reply:
Found a second place, where this was checked.